### ISSUE TITLE
Accessibility - aggregation of labels of children is not working.

### ISSUE THREAD

**Issue by anushreeag** [facebook/litho#443]
- [ YES] I have searched [existing issues](https://github.com/facebook/litho/issues) and this is not a duplicate

## Version

`  // Litho
    implementation 'com.facebook.litho:litho-core:0.15.0'
    implementation 'com.facebook.litho:litho-widget:0.15.0'
    compileOnly 'com.facebook.litho:litho-annotations:0.15.0'

    annotationProcessor 'com.facebook.litho:litho-processor:0.15.0'

    // SoLoader
    implementation 'com.facebook.soloader:soloader:0.2.0'`

## Issues and Steps to Reproduce
1. Enable talkback(accessibility) from settings.
2. Go to the sample Litho application.(Check the code below) 
3. Column should be focusable and individual text views should not be focusable and also text of text views should be aggregated. 

## Expected Behavior
In case of pure Android, 
if focusable is set on a ViewGroup( Ex : a linear layout) and this view group has 2 Text views, then in case of accessibility, 
1. Only Linear layout is focusable and text views are not focusable
2. The 2 text view's text is aggregated and talks back when linear layout is focused.
In case of accessibility Litho should behave same as Android.

## Link to Code
**_Android code_**
MainActivity.java
`public class MainActivity extends AppCompatActivity {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
      super.onCreate(savedInstanceState);
       setContentView(R.layout.activity_new);
  }

}`

activity_new.xml
![image](https://user-images.githubusercontent.com/17777328/46646094-132d9c80-cb3d-11e8-97d5-abaca85273bb.png)

**_Litho Code_** 
MainActivity.java
`public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LithoView view = new LithoView(this);
        ComponentContext context = view.getComponentContext();
        Component component = MyColumnComponent.create(context).focusable(true).build();
        setContentView(view.create(context, component));
    }
}`

MyColumnComponentSpec.java
`
@LayoutSpec
public class MyColumnComponentSpec {

  @OnCreateLayout
  static Component onCreateLayout(ComponentContext c) {

    return Column.create(c)
        .child(Text.create(c).text("Text1").textSizeDip(30).build())
        .child(Text.create(c).text("Text2").textSizeDip(30).build())
        .focusable(true)
        .build();
  }

}`

**Comment by anushreeag** [comment]
[LithoA11y.zip](https://github.com/facebook/litho/files/2458827/LithoA11y.zip)
[AndroidA11y.zip](https://github.com/facebook/litho/files/2458829/AndroidA11y.zip)

Attaching the source code of Litho and android projects tried for accessibility aggregation.

**Comment by vinc3m1** [comment]
Basically we're trying to replicate this behavior with Litho, which seems to be unsupported right now: https://developer.android.com/guide/topics/ui/accessibility/apps#grouping-content-organize

Do you all have any plans to implement this? Or any reason not to take a pull request if it's not prioritized?

**Comment by astreet** [comment]
I would want Brett who works on accessibility to weigh in -- I would almost definitely assume we'd take a PR :)

**Comment by appleguy** [comment]
Thanks Andy - if you could tag Brett or ask him offline, we'd definitely appreciate it! Although our team is stretched right now, some improvements here would make a difference in our ability to roll out to larger audiences.

**Comment by blavalla** [comment]
@appleguy - This is an unfortunate side effect from the fact that Litho was originally written against an older version of the Android Support library.

What is technically happening here is that any time Litho sees a Component that it deems necessary to create a View for accessibility, it sets an AccessibilityDelegate on it, which extends the "ExploreByTouchHelper" class  (see the code [here](https://github.com/facebook/litho/blob/d2541a376eef68f03781d3b4d2d4d00db787bbcc/litho-core/src/main/java/com/facebook/litho/ComponentHost.java#L956)).  This is a bit overkill, as you don't usually need the full functionality of this class, but it is convenient to use the same delegate no matter the reason the node was deemed to be necessary for accessibility.

In older versions of the support library the ExploreByTouchHelper class didn't actually modify the focusability of the View that it was attached to in its constructor (example [here](https://android.googlesource.com/platform/frameworks/support/+/f25dedc/v4/java/android/support/v4/widget/ExploreByTouchHelper.java#89)), but in newer versions it does (example [here](https://android.googlesource.com/platform/frameworks/support/+/71e4b67/v4/java/android/support/v4/widget/ExploreByTouchHelper.java#147)).  

What this means is that even though you aren't specifying "focusable(true)" on the Text components in your example, they are getting this property set on their created Views by the ExploreByTouchHelper class.  Since they are now focusable, the focusability of the parent with no description of its own, is correctly ignored.

The fix here is actually to not use the ExploreByTouchHelper class at all unless the view needs it for creating virtual child nodes.  I've been meaning to make this change for a while now, so let me see if I can get to it later this week.  If I can't get to it this week, I'll comment here again and if you want to make a pull request, I'd gladly review and accept the changes.

**Comment by appleguy** [comment]
@blavalla *thank you* for the informative comment! Several folks I work with are likewise very appreciative you weighed in here!

Although it sounds like a task for another day, I wonder if it would make sense to invest in creating Accessibility Node objects (hosted by a parent LithoView or ComponentHost) in order to avoid creating Views when the only properties triggering the View creation are a11y-related.

If it were done at the time that TalkBack actually queries for the nodes, it might not be too hard to walk the Component tree and build what's needed for Components that don't have their own views. That's the general approach that Texture uses [for this case](https://github.com/TextureGroup/Texture/blob/master/Source/Details/_ASDisplayViewAccessiblity.mm#L126), though it may not be appropriate for Litho (especially if the specifics of the implementation required would add a bunch of complexity).

In any case, we'd greatly appreciate your help if you do find the time to draft the PR (excluding ExploreByTouchHelper unless it's needed). No worries if it doesn't work out though, in which case we'll do our best shot at it. Cheers!

**Comment by blavalla** [comment]
@appleguy, just to keep you up to date here, I've submitted a diff that should resolve this issue.  I'm hoping to get it out this week.

The fix is actually much simpler than I originally envisioned, and essentially is just resetting the "focusable" and "importantForAccessibility" properties to their original values after ExploreByTouchHelper's constructor runs.

**Comment by vinc3m1** [comment]
thanks for the update @blavalla! 

does it look something like this:
```
    if (isAccessibilityEnabled && mComponentAccessibilityDelegate == null) {
      boolean focusable = isFocusable();
      mComponentAccessibilityDelegate = new ComponentAccessibilityDelegate(this);
      setFocusable(focusable);
    }
```

kudos to Jacek on our team for finding that one, we're currently using this internally, just haven't had time to submit a change upstream yet

**Comment by blavalla** [comment]
@vinc3m1, yep it looks almost exactly like that, with the only difference being the isFocusable result is passed into the ComponentAccessibilityDelegate's constructor and it's handled there, as well as also passing in the result of `ViewCompat.getImportantForAccessibility(this)`.  It also does the same thing when going through the descendants and refreshing their delegates. 

Glad you found a workaround to unblock you for now, and sorry for the slow turnaround on this one!

**Comment by anushreeag** [comment]
Thank you for fixing this issue. It works well now and as expected !

**Comment by blavalla** [comment]
@anushreeag sorry for the confusion, that comment was only relevant for our internal code base.  The one fix should be all you need for the open source version of Litho.  As far as I know, aggregation works as expected now, but feel free to open a new issue if you find any new problems with it.

Thanks!

**Comment by anushreeag** [comment]
I have tested the fix by building a sample Litho App with 2 text components . Aggregation works well as expected now.
Thanks for fixing this issue.
